Doc/library/importlib.rst

```diff
@@ -1284,66 +1307,101 @@ find and load modules.
+   The place from which the module's data was imported.  Its value and meaning
+   is up to the finder.
+   It is the relative path for packages run from the file system and the
+   absolute path for imported modules and modules run from the module
+   namespace.
+   It is ``None`` for imported namespace packages, ``'built-in'`` for imported
+   built-in modules and ``'frozen'`` for imported frozen modules.
+   It is not set for non-package modules run from the file system and
+   non-package modules run from standard input.
 
    .. attribute:: submodule_search_locations
 
    (``__path__``)
 
-   List of strings for where to find submodules, if a package (``None``
-   otherwise).
+   The (possibly empty) iterable of all locations to search (i.e. the search
+   path) when looking for the package's submodules.  Order is significant.
+   It is ``None`` for packages run from the file system, imported non-package
+   modules and modules run from the module namespace.
+   It is not set for non-package modules run from the file system and
+   non-package modules run from standard input.
 
    .. attribute:: loader_state
 
-   Container of extra module-specific data for use during loading (or
-   ``None``).
+   The finder may set this to an object containing additional data for the
+   loader to use when loading the module.  Otherwise it will be ``None``.
 
    .. attribute:: cached
 
    (``__cached__``)
 
-   String for where the compiled module should be stored (or ``None``).
+   The location of the module's cached data.  Some loaders support optimization
+   through caching the bytecode of the module.  ``cached`` is how the finder
+   tells the loader where to look for that cached bytecode, or where to store
+   it.
+   It is the relative path for packages run from the file system and the
+   absolute path for imported modules and modules run from the module
+   namespace.
+   It is ``None`` for imported namespace packages, imported built-in modules
+   and imported frozen modules.
+   It is not set for non-package modules run from the file system and
+   non-package modules run from standard input.
 
    .. attribute:: parent
 
    (``__package__``)
 
-   (Read-only) Fully-qualified name of the package to which the module
-   belongs as a submodule (or ``None``).
+   The module's ``__spec.__name`` attribute for imported packages and packages
+   run from the module namespace, the parent package's ``__spec__.name``
+   attribute for imported non-package submodules and non-paclage submodules run
+   from the module namespace and ``''`` for packages run from the file system,
+   imported non-package top-level modules and non-package top-level modules run
+   from the module namespace.
+   It is not set for non-package modules run from the file system and
+   non-package modules run from standard input.
+   It is a read-only attribute.
 
    .. attribute:: has_location
 
-   Boolean indicating whether or not the module's "origin"
```

Comment:
```
   True if the spec's :attr:`origin` refers to a loadable location.  False otherwise.
   This value impacts how ``origin`` is interpreted and how the module's
   ``__file__`` is populated.
```

Reply:
Alright.